docs/source/python/compute.rst

```diff
@@ -292,4 +292,70 @@ The resulting dataset will be an :class:`.InMemoryDataset` containing the joined
    animal: [["Brittle stars",null,null]]
    n_legs: [[5,null,null]]
 
+.. _py-filter-expr:
 
+Filtering by Expressions
+========================
+
+:class:`.Table` and :class:`.Dataset` can
+both be filtered using a boolean :class:`.Expression`.
+
+The expression can be built starting from a 
+:func:`pyarrow.field`, comparisons and transformations
+can then be applied to one or more fields to build the filter
+expression you care about.
+
+Most :ref:`compute` can be used to perform transformations
+on a ``field``.
+
+For example we could build a filter to find all rows that are even
+in column ``"nums"``
+
+.. code-block:: python
+
+   even_filter = (pc.bit_wise_and(pc.field("nums"), pc.scalar(1)) == pc.scalar(0))
```

Comment:
Many examples in this section repeat the import statements. I don't think it's necessary to do this for _every_ code block, but maybe worth doing in the first block of this section. That would be helpful for readers who skipped to this section and aren't yet familiar with `pc` as `pyarrow.compute`.

Reply:
That's a good catch, added explicit import